internal/util/pod_security.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2017 - 2025 Crunchy Data Solutions, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/crunchydata/postgres-operator/internal/initialize"
+)
+
+// PodSecurityContext returns a v1.PodSecurityContext for cluster that can write
+// to PersistentVolumes.
+func PodSecurityContext(fsgroup int64, supplementalGroups []int64, openshift bool) *corev1.PodSecurityContext {
```

Comment:
🤔 I think I'd prefer the caller to pass in `fsgroup = 0` when they know its OpenShift, but a single place with this knowledge is good, too.

🤔 🤔 PgBouncer is deployed from a PostgresCluster spec which has an isOpenShift override.

Reply:
on PodSecurityContext, fsgroup is a pointer (with omitempty) so we don't want to set it at all, right?